scrapy/http/response/__init__.py

```diff
@@ -110,65 +132,65 @@ def _set_body(self, body):
 
     body = property(_get_body, obsolete_setter(_set_body, "body"))
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"<{self.status} {self.url}>"
 
-    def copy(self):
+    def copy(self) -> Response:
```

Comment:
Response typing seems wrong for subclasses

Reply:
Same `cls`-related problem.